lnd_services.go

```diff
@@ -533,6 +565,60 @@ func (s *GrpcLndServices) waitForChainSync(ctx context.Context,
 	return <-update
 }
 
+// waitForChainNotifier blocks until the ChainNotifier RPC accepts block epoch
+// subscriptions and delivers at least one block height.
+func (s *GrpcLndServices) waitForChainNotifier(ctx context.Context,
+	timeout, pollInterval time.Duration) error {
+
+	mainCtx := ctx
+	if mainCtx == nil {
+		mainCtx = context.Background()
+	}
+
+	register := s.ChainNotifier.RegisterBlockEpochNtfn
+
+	for {
+		// Make new RegisterBlockEpochNtfn call.
+		subCtx, cancel := context.WithTimeout(mainCtx, timeout)
+		blockChan, errChan, err := register(subCtx)
+		if err != nil {
+			cancel()
+
+			return fmt.Errorf("register block epoch ntfn: %w", err)
+		}
+
+		// Wait for block height notification, which indicates success.
+		select {
+		case <-mainCtx.Done():
+			cancel()
+
+			return mainCtx.Err()
+
+		case err := <-errChan:
+			cancel()
+
+			// If chainNotifier is not ready yet, retry.
+			if isChainNotifierStartingErr(err) {
+				select {
+				case <-time.After(pollInterval):
+					continue
```

Comment:
Without having tested this myself on the first iteration, I just want to make sure that something is logged here or during the `register` call. Just so that the user sees that there's still some progress and that the application isn't frozen (similar to how the `waitForChainSync` function triggers logging of every `GetInfo` poll attempt.).